base_geoengine/fields.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2011-2012 Nicolas Bessi (Camptocamp SA)
+# Copyright 2016 Yannick Payot (Camptocamp SA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+from operator import attrgetter
+
+from odoo import _, fields
+from odoo.tools import sql
+
+from .geo_db import create_geo_column
+from .geo_helper import geo_convertion_helper as convert
+
+logger = logging.getLogger(__name__)
+try:
+    import geojson
+    from shapely.geometry import Point
+    from shapely.geometry.base import BaseGeometry
+    from shapely.wkb import loads as wkbloads
+except ImportError:
+    logger.warning("Shapely or geojson are not available in the sys path")
+
+
+class GeoField(fields.Field):
+    """The field descriptor contains the field definition common to all
+    specialized fields for geolocalization. Subclasses must define a type
+    and a geo_type. The type is the name of the corresponding column type,
+    the geo_type is the name of the corresponding type in the GIS system.
+    """
+
+    geo_type = None
+    dim = 2
+    srid = 3857
+    gist_index = True
+
+    @property
+    def column_format(self):
+        return "ST_GeomFromText(%s, {})".format(self.srid)
+
+    @property
+    def column_type(self):
+        return ("geometry", "geometry")
+
+    def convert_to_column(self, value, record, values=None):
+        """Convert value to database format
+
+        value can be geojson, wkt, shapely geometry object.
+        If geo_direct_write in context you can pass diretly WKT"""
+        if not value:
+            return None
+        shape_to_write = self.entry_to_shape(value, same_type=True)
+        if shape_to_write.is_empty:
+            return None
+        else:
+            return shape_to_write.wkt
```

Comment:
@lmignon , @kouffsamuel

I am getting an error when editing a geoField,

```sql
UPDATE "res_partner" SET "geo_point" = 'POINT (-13578158.298793 4513301.6464)';
```
```log
psycopg2.errors.InvalidParameterValue: Geometry SRID (0) does not match column SRID (3857)
```
I fixed it by setting the SRID for the given point 

 ```suggestion
            return f'SRID={self.srid};{shape_to_write.wkt}'
```

Reply:
@kouffsamuel @sbejaoui This conversion was previously implemented in 10.0 by the use of the 'column_format' on the field definition https://github.com/OCA/geospatial/blob/10.0/base_geoengine/fields.py#L36 
```python
    @property
    def column_format(self):
        return "ST_GeomFromText(%%s, %s)" % (self.srid,)
```

The `convert_to_column` method seems to be the replacement of the `column_format` method.  Would it be more accurate to write?
```suggestion
            return f"ST_GeomFromText({shape_to_write.wkt}, {self.srid})"
```